packages/search-input/src/SearchInput.spec.tsx

```diff
@@ -30,6 +37,47 @@ describe('packages/search-input', () => {
     expect(searchInput.getAttribute('type')).toBe('search');
   });
 
+  test(`renders ${defaultProps.placeholder} as placeholder text`, () => {
+    const { getByPlaceholderText } = renderSearchInput(defaultProps);
+    expect(getByPlaceholderText(defaultProps.placeholder)).toBeVisible();
+  });
+
+  test(`renders ${defaultProps.className} in the classList`, () => {
+    const { container } = renderSearchInput(defaultProps);
+    expect(
+      (container?.firstChild as HTMLElement)?.classList.contains(
+        defaultProps.className,
+      ),
+    ).toBe(true);
+  });
+
+  test('key presses are reflected in component and onChange function is called when value changes', () => {
+    const { searchInput } = renderSearchInput(defaultProps);
+    expect((searchInput as HTMLInputElement).value).toBe('');
+
+    fireEvent.change(searchInput, {
+      target: { value: 'a' },
+    });
```

Comment:
I think we should prefer using `userEvent.type` to more closely simulate real user behavior